src/include/duckdb/common/types.hpp

```diff
@@ -436,8 +436,13 @@ struct LogicalType {
 	DUCKDB_API bool IsIntegral() const;
 	DUCKDB_API bool IsNumeric() const;
 	DUCKDB_API hash_t Hash() const;
+	DUCKDB_API void SetExtraInfo(shared_ptr<ExtraTypeInfo> type_info);
```

Comment:
Is this method necessary? `extra_info` is supposed to be an internal property, do we need a public method that modifies the extra info?